<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='clinvar_plot() integrates ClinVar classifications
with AlphaMissense predicted scores derived from
am_data("aa_substitutions") and returns a ggplot object for
visualization.
clinvar_data() loads in the raw ClinVar information from
the supplemental table of the AlphaMissense
[2023] paper.'><title>Integrate ClinVar Labels with AlphaMissense Pathogenicity Scores — clinvar_plot • AlphaMissenseR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Integrate ClinVar Labels with AlphaMissense Pathogenicity Scores — clinvar_plot"><meta property="og:description" content='clinvar_plot() integrates ClinVar classifications
with AlphaMissense predicted scores derived from
am_data("aa_substitutions") and returns a ggplot object for
visualization.
clinvar_data() loads in the raw ClinVar information from
the supplemental table of the AlphaMissense
[2023] paper.'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">AlphaMissenseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/a_introduction.html">Accessing AlphaMissense Resources in R</a>
    <a class="dropdown-item" href="../articles/b_visualization.html">Visualization</a>
    <a class="dropdown-item" href="../articles/c_issues.html">Issues &amp; Solutions</a>
    <a class="dropdown-item" href="../articles/d_integrate_ClinVar.html">Integration with ClinVar</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/mtmorgan/AlphaMissenseR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Integrate ClinVar Labels with AlphaMissense Pathogenicity Scores</h1>
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/AlphaMissenseR/blob/HEAD/R/clinvar_plot.R" class="external-link"><code>R/clinvar_plot.R</code></a></small>
      <div class="d-none name"><code>clinvar_plot.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>clinvar_plot()</code> integrates ClinVar classifications
with AlphaMissense predicted scores derived from
<code>am_data("aa_substitutions")</code> and returns a ggplot object for
visualization.</p>
<p><code>clinvar_data()</code> loads in the raw ClinVar information from
the supplemental table of the AlphaMissense
<a href="https://www.science.org/doi/10.1126/science.adg7492" class="external-link">[2023]</a> paper.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">clinvar_plot</span><span class="op">(</span><span class="va">uniprotId</span>, <span class="va">alphamissense_table</span>, <span class="va">clinvar_table</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">clinvar_data</span><span class="op">(</span>record <span class="op">=</span> <span class="va">ALPHAMISSENSE_RECORD</span>, bfc <span class="op">=</span> <span class="fu">BiocFileCache</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>uniprotId</dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> a valid UniProt accession identifier.</p></dd>


<dt>alphamissense_table</dt>
<dd><p>a table containing AlphaMissense
predictions for protein variants. By default, the table is
derived from <code>am_data("aa_substitution")</code>. Alternatively, a
user-defined <code><a href="https://tibble.tidyverse.org/reference/tbl_df-class.html" class="external-link">tibble::tbl_df</a></code> or <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></code>
can be supplied.</p></dd>


<dt>clinvar_table</dt>
<dd><p>a table containing ClinVar information. By
default, the table is derived from the supplemental data of the
AlphaMissense paper. Alternatively, a user-defined
<code><a href="https://tibble.tidyverse.org/reference/tbl_df-class.html" class="external-link">tibble::tbl_df</a></code> or <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></code> can be
supplied.</p></dd>


<dt>record</dt>
<dd><p>character(1) Zenodo record for the AlphaMissense data
resources.</p></dd>


<dt>bfc</dt>
<dd><p>an object returned by <code>BiocFileCache()</code> representing the
location of the AlphaMissenseR database. The default is the
'global' BiocFileCache.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p><code>clinvar_plot()</code> returns a <code>ggplot</code> object which overlays
ClinVar classifications onto AlphaMissense predicted
scores. Blue, gray, and red colors represent pathogenicity
classifications for "likely benign", "ambiguous", or
"likely pathogenic", respectively. Large, bolded points are
ClinVar variants colored according to their clinical
classification, while smaller points in the background are
AlphaMissense predictions.</p>


<p><code>clinvar_data()</code> returns a tbl with 82872 rows and 5 variables:</p><ul><li><p><code>variant_id</code>: ClinVar variant identifier.</p></li>
<li><p><code>transcript_id</code>: Ensembl transcript identifier.</p></li>
<li><p><code>protein_variant</code>: UniProt accession:protein variant identifier.</p></li>
<li><p><code>AlphaMissense</code>: AlphaMissense pathogenicity score.</p></li>
<li><p><code>label</code>: Binary ClinVar class, either "benign" or "pathogenic".</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>For <code>alphamissense_table</code>, columns must include:</p><ul><li><p><code>uniprot_id</code>: UniProt accession identifier.</p></li>
<li><p><code>protein_variant</code>: variant identifier string, with protein
position in the middle, and the reference and mutant amino acid
residues to the left and right of the position, respectively.</p></li>
<li><p><code>am_class</code>: AlphaMissense classification of either "benign",
"ambiguous", or "pathogenic".</p></li>
<li><p><code>am_pathogenicity</code>: AlphaMissense predicted score.</p></li>
</ul><p>For <code>clinvar_table</code>, columns must include:</p><ul><li><p><code>uniprot_id</code>: UniProt accession identifier, matching <code>alphamissense_table</code>.</p></li>
<li><p><code>protein_variant</code>: variant identifier string, matching
<code>alphamissense_table</code> format.</p></li>
<li><p><code>cv_class</code>: binary ClinVar classification of "benign" or "pathogenic".</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Cheng et al.,
Accurate proteome-wide missense variant effect prediction with AlphaMissense.
<em>Science</em> 381, eadg7492. DOI:10.1126/science.adg7492.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">alphamissense_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="AlphaMissense.html">am_data</a></span><span class="op">(</span><span class="st">"aa_substitutions"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">clinvar_plot</span><span class="op">(</span>uniprotId <span class="op">=</span> <span class="st">"P37023"</span>,</span></span>
<span class="r-in"><span>   alphamissense_table <span class="op">=</span> <span class="va">alphamissense_table</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * [15:32:19][info] 'clinvar_table' not provided, using default ClinVar dataset in AlphaMissenseR package</span>
<span class="r-plt img"><img src="clinvar_plot-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martin Morgan, Tram Nguyen, Chan Zuckerberg Initiative DAF CZF2019-002443, NIH NCI ITCR U24CA180996, NIH NCI IOTN U24CA232979, NIH NCI ARTNet U24CA274159.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

