<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Integrate ClinVar Labels with AlphaMissense Pathogenicity Scores — clinvar_data • AlphaMissenseR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Integrate ClinVar Labels with AlphaMissense Pathogenicity Scores — clinvar_data"><meta name="description" content='clinvar_data() loads ClinVar information from the
supplemental table of the AlphaMissense
[2023]
paper.
clinvar_plot() integrates ClinVar classifications
with AlphaMissense predicted scores derived from
am_data("aa_substitutions") and returns a ggplot object for
visualization.'><meta property="og:description" content='clinvar_data() loads ClinVar information from the
supplemental table of the AlphaMissense
[2023]
paper.
clinvar_plot() integrates ClinVar classifications
with AlphaMissense predicted scores derived from
am_data("aa_substitutions") and returns a ggplot object for
visualization.'></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">AlphaMissenseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.7</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/introduction.html">A. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/alphafold.html">B. AlphaFold Integration</a></li>
    <li><a class="dropdown-item" href="../articles/clinvar.html">C. ClinVar Integration</a></li>
    <li><a class="dropdown-item" href="../articles/benchmarking.html">D. Benchmarking with ProteinGym</a></li>
    <li><a class="dropdown-item" href="../articles/issues.html">E. Issues &amp; Solutions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mtmorgan/AlphaMissenseR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Integrate ClinVar Labels with AlphaMissense Pathogenicity Scores</h1>
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/AlphaMissenseR/blob/HEAD/R/ClinVar.R" class="external-link"><code>R/ClinVar.R</code></a></small>
      <div class="d-none name"><code>ClinVar.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>clinvar_data()</code> loads ClinVar information from the
supplemental table of the AlphaMissense
<a href="https://www.science.org/doi/10.1126/science.adg7492" class="external-link">[2023]</a>
paper.</p>
<p><code>clinvar_plot()</code> integrates ClinVar classifications
with AlphaMissense predicted scores derived from
<code>am_data("aa_substitutions")</code> and returns a ggplot object for
visualization.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">clinvar_data</span><span class="op">(</span>record <span class="op">=</span> <span class="va">ALPHAMISSENSE_RECORD</span>, bfc <span class="op">=</span> <span class="fu">BiocFileCache</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">clinvar_plot</span><span class="op">(</span><span class="va">uniprotId</span>, <span class="va">alphamissense_table</span>, <span class="va">clinvar_table</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-record">record<a class="anchor" aria-label="anchor" href="#arg-record"></a></dt>
<dd><p>character(1) Zenodo record for the AlphaMissense data
resources.</p></dd>


<dt id="arg-bfc">bfc<a class="anchor" aria-label="anchor" href="#arg-bfc"></a></dt>
<dd><p>an object returned by <code>BiocFileCache()</code> representing the
location of the AlphaMissenseR database. The default is the
'global' BiocFileCache.</p></dd>


<dt id="arg-uniprotid">uniprotId<a class="anchor" aria-label="anchor" href="#arg-uniprotid"></a></dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> a valid UniProt accession identifier.</p></dd>


<dt id="arg-alphamissense-table">alphamissense_table<a class="anchor" aria-label="anchor" href="#arg-alphamissense-table"></a></dt>
<dd><p>a table containing AlphaMissense
predictions for protein variants. By default, the table is
derived from <code>am_data("aa_substitution")</code>. Alternatively, a
user-defined <code><a href="https://tibble.tidyverse.org/reference/tbl_df-class.html" class="external-link">tibble::tbl_df</a></code> or <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></code>
can be supplied.</p></dd>


<dt id="arg-clinvar-table">clinvar_table<a class="anchor" aria-label="anchor" href="#arg-clinvar-table"></a></dt>
<dd><p>a table containing ClinVar information. By
default, the table is derived from the supplemental data of the
AlphaMissense paper. Alternatively, a user-defined
<code><a href="https://tibble.tidyverse.org/reference/tbl_df-class.html" class="external-link">tibble::tbl_df</a></code> or <code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></code> can be
supplied.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><code>clinvar_data()</code> returns a tbl with 82872 rows and 5 variables:</p><ul><li><p><code>variant_id</code>: ClinVar variant identifier.</p></li>
<li><p><code>transcript_id</code>: Ensembl transcript identifier.</p></li>
<li><p><code>protein_variant</code>: UniProt accession:protein variant identifier.</p></li>
<li><p><code>AlphaMissense</code>: AlphaMissense pathogenicity score.</p></li>
<li><p><code>label</code>: Binary ClinVar class, either "benign" or "pathogenic".</p></li>
</ul><p><code>clinvar_plot()</code> returns a <code>ggplot</code> object which overlays
ClinVar classifications onto AlphaMissense predicted
scores. Blue, gray, and red colors represent pathogenicity
classifications for "likely benign", "ambiguous", or
"likely pathogenic", respectively. Large, bolded points are
ClinVar variants colored according to their clinical
classification, while smaller points in the background are
AlphaMissense predictions.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>For <code>clinvar_plot()</code>,  <code>alphamissense_table</code> columns must include:</p><ul><li><p><code>uniprot_id</code>: UniProt accession identifier.</p></li>
<li><p><code>protein_variant</code>: variant identifier string, with protein
position in the middle, and the reference and mutant amino acid
residues to the left and right of the position, respectively.</p></li>
<li><p><code>am_class</code>: AlphaMissense classification of either "benign",
"ambiguous", or "pathogenic".</p></li>
<li><p><code>am_pathogenicity</code>: AlphaMissense predicted score.</p></li>
</ul><p><code>clinvar_table</code> columns must include:</p><ul><li><p><code>uniprot_id</code>: UniProt accession identifier, matching <code>alphamissense_table</code>.</p></li>
<li><p><code>protein_variant</code>: variant identifier string, matching
<code>alphamissense_table</code> format.</p></li>
<li><p><code>cv_class</code>: binary ClinVar classification of "benign" or "pathogenic".</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Cheng et al.,
Accurate proteome-wide missense variant effect prediction with AlphaMissense.
<em>Science</em> 381, eadg7492. DOI:10.1126/science.adg7492.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">clinvar_data</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># Source:   table&lt;clinvar&gt; [?? x 5]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># Database: DuckDB v1.1.0 [root@Darwin 23.6.0:R 4.5.0//Users/mtmorgan/Library/Caches/org.R-project.R/R/BiocFileCache/121787f1dafbc_121787f1dafbc]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id           transcript_id     protein_variant AlphaMissense label </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> chr1_925969_C_T_hg38 ENST00000342066.8 Q96NU1:P10S            0.967  benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> chr1_930165_G_A_hg38 ENST00000342066.8 Q96NU1:R28Q            0.663  benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> chr1_930204_G_A_hg38 ENST00000342066.8 Q96NU1:R41Q            0.086<span style="text-decoration: underline;">6</span> benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> chr1_930245_G_A_hg38 ENST00000342066.8 Q96NU1:D55N            0.134  benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> chr1_930248_G_A_hg38 ENST00000342066.8 Q96NU1:G56S            0.100  benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> chr1_930282_G_A_hg38 ENST00000342066.8 Q96NU1:R67Q            0.063<span style="text-decoration: underline;">5</span> benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> chr1_930285_G_A_hg38 ENST00000342066.8 Q96NU1:R68Q            0.062<span style="text-decoration: underline;">9</span> benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> chr1_930314_C_T_hg38 ENST00000342066.8 Q96NU1:H78Y            0.110  benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> chr1_930320_C_T_hg38 ENST00000342066.8 Q96NU1:R80C            0.091<span style="text-decoration: underline;">8</span> benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> chr1_931058_G_A_hg38 ENST00000342066.8 Q96NU1:V92M            0.196  benign</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">alphamissense_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="AlphaMissense.html">am_data</a></span><span class="op">(</span><span class="st">"aa_substitutions"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">clinvar_plot</span><span class="op">(</span></span></span>
<span class="r-in"><span>    uniprotId <span class="op">=</span> <span class="st">"P37023"</span>,</span></span>
<span class="r-in"><span>    alphamissense_table <span class="op">=</span> <span class="va">alphamissense_table</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * [12:02:05][info] 'clinvar_table' not provided, using default ClinVar dataset in AlphaMissenseR package</span>
<span class="r-plt img"><img src="ClinVar-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martin Morgan, Tram Nguyen, Chan Zuckerberg Initiative DAF CZF2019-002443, NIH NCI ITCR U24CA180996, NIH NCI IOTN U24CA232979, NIH NCI ARTNet U24CA274159.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

