<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>D. Benchmarking with ProteinGym • AlphaMissenseR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="D. Benchmarking with ProteinGym">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">AlphaMissenseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.7</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/introduction.html">A. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/alphafold.html">B. AlphaFold Integration</a></li>
    <li><a class="dropdown-item" href="../articles/clinvar.html">C. ClinVar Integration</a></li>
    <li><a class="dropdown-item" href="../articles/benchmarking.html">D. Benchmarking with ProteinGym</a></li>
    <li><a class="dropdown-item" href="../articles/issues.html">E. Issues &amp; Solutions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mtmorgan/AlphaMissenseR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>D. Benchmarking with ProteinGym</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/AlphaMissenseR/blob/HEAD/vignettes/benchmarking.Rmd" class="external-link"><code>vignettes/benchmarking.Rmd</code></a></small>
      <div class="d-none name"><code>benchmarking.Rmd</code></div>
    </div>

    
    
<p>Original version: 8 October, 2024</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mtmorgan.github.io/AlphaMissenseR/">AlphaMissenseR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mjskay.github.io/ggdist/" class="external-link">ggdist</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/erocoar/gghalves" class="external-link">gghalves</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Benchmarking is essential for assessing different variant effect
prediction approaches, potentially aiding in decisions about which
models are most suitable for specific research questions or clinical
applications.</p>
<p>To evaluate the performance of AlphaMissense and its predictions, we
integrate with the <a href="https://proteingym.org/" class="external-link">ProteinGym</a>
database, a comprehensive collection of benchmarks aimed at comparing
the ability of models that predict the effects of protein mutations. It
consists of an extensive dataset of deep mutational scanning (DMS)
assays covering approximately 2.7 million missense variants across 217
experiments, as well as annotated human clinical variants for 2,525
proteins. DMS assays are highly valuable as they systematically test all
possible single mutations in a protein, recording their fitness effects.
They can reveal the impacts of both deleterious and beneficial mutations
on fitness, offering insights into protein structure and activity.</p>
<p>Furthermore, ProteinGym provides several standardized model
evaluation metric scores (“AUC”, “MCC”, “NDCG”, “Spearman”,
“Top_recall”) for 62 models calculated across the 217 DMS assays,
offering a consistent and extensive benchmarking framework across
approaches.</p>
<p>This vignette demonstrates how to 1) compare AlphaMissense
predictions to DMS scores on a per-protein basis, and 2) how to
benchmark AlphaMissense against other models aimed at predicting protein
mutation effects.</p>
</div>
<div class="section level2">
<h2 id="access-proteingym-datasets-through-experimenthub-">Access ProteinGym datasets through <code>ExperimentHub</code>.<a class="anchor" aria-label="anchor" href="#access-proteingym-datasets-through-experimenthub-"></a>
</h2>
<p>The ProteinGym DMS assays can be accessed by querying
<code>ExperimentHub</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">dms_data</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">'EH9555'</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; ProteinGymR not installed.</span></span>
<span><span class="co">#&gt;   Full functionality, documentation, and loading of data might not be possible without installing</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dms_data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "A0A140D2T1_ZIKV_Sourisseau_2019"  "A0A192B1T2_9HIV1_Haddox_2018"    </span></span>
<span><span class="co">#&gt; [3] "A0A1I9GEU1_NEIME_Kennouche_2019"  "A0A247D711_LISMN_Stadelmann_2021"</span></span>
<span><span class="co">#&gt; [5] "A0A2Z5U3Z0_9INFA_Doud_2016"       "A0A2Z5U3Z0_9INFA_Wu_2014"</span></span></code></pre></div>
<p>Each element of the list is an individual DMS assay with the
following information for each column: the UniProt protein identifier,
the DMS experiment assay identifier, the mutant at a given protein
position, the mutated protein sequence, the recorded DMS score, and a
binary DMS score bin categorizing whether the mutation has high (1) or
low fitness (0). For more details, reference the publication from Notin
et al. <a href="https://papers.nips.cc/paper_files/paper/2023/hash/cac723e5ff29f65e3fcbb0739ae91bee-Abstract-Datasets_and_Benchmarks.html" class="external-link">2023</a>.</p>
<p>A supplementary table of AlphaMissense pathogencity scores for ~1.6 M
substitutions matching those in the ProteinGym DMS assays is provided by
Cheng et al. <a href="https://www.science.org/doi/10.1126/science.adg7492" class="external-link">2023</a>, and
can also be accessed through <code>ExperimentHub</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">am_scores</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">'EH9554'</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; ProteinGymR not installed.</span></span>
<span><span class="co">#&gt;   Full functionality, documentation, and loading of data might not be possible without installing</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">am_scores</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                   DMS_id      Uniprot_ID variant_id</span></span>
<span><span class="co">#&gt; 1 A0A140D2T1_ZIKV_Sourisseau_growth_2019 A0A140D2T1_ZIKV      I291A</span></span>
<span><span class="co">#&gt; 2 A0A140D2T1_ZIKV_Sourisseau_growth_2019 A0A140D2T1_ZIKV      I291Y</span></span>
<span><span class="co">#&gt; 3 A0A140D2T1_ZIKV_Sourisseau_growth_2019 A0A140D2T1_ZIKV      I291W</span></span>
<span><span class="co">#&gt; 4 A0A140D2T1_ZIKV_Sourisseau_growth_2019 A0A140D2T1_ZIKV      I291V</span></span>
<span><span class="co">#&gt; 5 A0A140D2T1_ZIKV_Sourisseau_growth_2019 A0A140D2T1_ZIKV      I291T</span></span>
<span><span class="co">#&gt; 6 A0A140D2T1_ZIKV_Sourisseau_growth_2019 A0A140D2T1_ZIKV      I291S</span></span>
<span><span class="co">#&gt;   AlphaMissense</span></span>
<span><span class="co">#&gt; 1     0.9124037</span></span>
<span><span class="co">#&gt; 2     0.8899629</span></span>
<span><span class="co">#&gt; 3     0.9694002</span></span>
<span><span class="co">#&gt; 4     0.2470229</span></span>
<span><span class="co">#&gt; 5     0.8939576</span></span>
<span><span class="co">#&gt; 6     0.8617630</span></span></code></pre></div>
<p>The <code>data.frame</code> shows the <code>DMS_id</code> matching a
ProteinGym assay, the UniProt entry name of the protein evaluated, the
mutation and position, and the aggregated AlphaMissense score for that
mutation. For more details about the table, reference the <a href="https://www.science.org/doi/10.1126/science.adg7492" class="external-link">AlphaMissense</a>
paper.</p>
</div>
<div class="section level2">
<h2 id="correlate-dms-scores-and-alphamissense-predictions">Correlate DMS scores and AlphaMissense predictions<a class="anchor" aria-label="anchor" href="#correlate-dms-scores-and-alphamissense-predictions"></a>
</h2>
<p>The DMS scores serve as an experimental measure to compare the
accuracy of AlphaMissense mutation effect predictions. For a given
protein, we can plot the relationship between the two measures and
report their Spearman correlation.</p>
<p>Here, we demonstrate using the “NUD15_HUMAN” assay. First, we filter
both datasets to the chosen assay.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NUD15_dms</span> <span class="op">&lt;-</span> <span class="va">dms_data</span><span class="op">[[</span><span class="st">"NUD15_HUMAN_Suiter_2020"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">NUD15_am</span> <span class="op">&lt;-</span> <span class="va">am_scores</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">DMS_id</span> <span class="op">==</span> <span class="st">"NUD15_HUMAN_Suiter_2020"</span><span class="op">)</span></span></code></pre></div>
<p>Wrangle and merge the DMS and AlphaMissense tables together by the
UniProt and mutant identifers.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NUD15_am</span> <span class="op">&lt;-</span> <span class="va">NUD15_am</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Uniprot_ID <span class="op">=</span> <span class="st">"Q9NV35"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">merged_table</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="va">NUD15_am</span>, <span class="va">NUD15_dms</span>, </span>
<span>        by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Uniprot_ID"</span> <span class="op">=</span> <span class="st">"UniProt_id"</span>, <span class="st">"variant_id"</span> <span class="op">=</span> <span class="st">"mutant"</span><span class="op">)</span>,</span>
<span>        relationship <span class="op">=</span> <span class="st">"many-to-many"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Uniprot_ID</span>, <span class="va">variant_id</span>, <span class="va">AlphaMissense</span>, <span class="va">DMS_score</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Now, we plot the correlation.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">correlation_plot</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">merged_table</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">AlphaMissense</span>, x <span class="op">=</span> <span class="va">.data</span><span class="op">$</span><span class="va">DMS_score</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bin_2d.html" class="external-link">geom_bin2d</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">60</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html" class="external-link">scale_fill_continuous</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"DMS score"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"AlphaMissense score"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, vjust <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span>, vjust <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_plot</span></span></code></pre></div>
<p><img src="benchmarking_files/figure-html/dms_am_plot-1.png" width="480"></p>
<p>If mutating a residue resulted in reduced fitness in the DMS assay
(low DMS score), we would expect it to be predicted as pathogenic
(higher AlphaMissense score). Therefore, a stronger negative correlation
represents a tighter relationship between the two measures. We can see
this with the NUD15 protein.</p>
<p>We can also print the Spearman correlation.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span></span>
<span>    <span class="va">merged_table</span><span class="op">$</span><span class="va">AlphaMissense</span>, <span class="va">merged_table</span><span class="op">$</span><span class="va">DMS_score</span>, </span>
<span>    method<span class="op">=</span><span class="st">"spearman"</span>, </span>
<span>    exact <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  Spearman's rank correlation rho</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; data:  merged_table$AlphaMissense and merged_table$DMS_score</span></span>
<span><span class="co">#&gt; S = 6440239302, p-value &lt; 2.2e-16</span></span>
<span><span class="co">#&gt; alternative hypothesis: true rho is not equal to 0</span></span>
<span><span class="co">#&gt; sample estimates:</span></span>
<span><span class="co">#&gt;        rho </span></span>
<span><span class="co">#&gt; -0.6798269</span></span></code></pre></div>
<p>The correlation is r = -0.67 and is statistically significant.</p>
</div>
<div class="section level2">
<h2 id="benchmark-alphamissense-with-other-variant-effect-prediction-models">Benchmark AlphaMissense with other variant effect prediction
models<a class="anchor" aria-label="anchor" href="#benchmark-alphamissense-with-other-variant-effect-prediction-models"></a>
</h2>
<p>We can calculate the Spearman correlation across multiple DMS assays
in which we have corresponding AlphaMissense scores, and use this metric
to benchmark across multiple models aimed at predicting mutation
effects. For this analysis, we will load in the metric scores for 62
models calculated across the 217 DMS assays from the ProteinGym database
available through <code>ExperimentHub</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metrics_scores</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">'EH9593'</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; ProteinGymR not installed.</span></span>
<span><span class="co">#&gt;   Full functionality, documentation, and loading of data might not be possible without installing</span></span>
<span><span class="co">#&gt; loading from cache</span></span></code></pre></div>
<p><code>metrics_scores</code> is a <code>list</code> object where each
element is a <code>data.frame</code> corresponding to one of five model
evaluation metrics (“AUC”, “MCC”, “NDCG”, “Spearman”, “Top_recall”)
available from ProteinGym. Briefly, these metrics were calculated on the
DMS assays for 62 models in a zero-shot setting. The <a href="https://papers.nips.cc/paper_files/paper/2023/hash/cac723e5ff29f65e3fcbb0739ae91bee-Abstract-Datasets_and_Benchmarks.html" class="external-link">Protein
Gym paper</a> provides more information about these metrics.</p>
<p>To demonstrate, we will benchmarking using the Spearman correlation
calculated on 10 DMS assays for 5 models. The following code subsets and
merges the relevant datasets.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chosen_assays</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"A0A1I9GEU1_NEIME_Kennouche_2019"</span>, </span>
<span>    <span class="st">"A0A192B1T2_9HIV1_Haddox_2018"</span>, </span>
<span>    <span class="st">"ADRB2_HUMAN_Jones_2020"</span>, </span>
<span>    <span class="st">"BRCA1_HUMAN_Findlay_2018"</span>, </span>
<span>    <span class="st">"CALM1_HUMAN_Weile_2017"</span>,</span>
<span>    <span class="st">"GAL4_YEAST_Kitzman_2015"</span>, </span>
<span>    <span class="st">"Q59976_STRSQ_Romero_2015"</span>, </span>
<span>    <span class="st">"UBC9_HUMAN_Weile_2017"</span>, </span>
<span>    <span class="st">"TPK1_HUMAN_Weile_2017"</span>,</span>
<span>    <span class="st">"YAP1_HUMAN_Araya_2012"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">am_subset</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">am_scores</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">DMS_id</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">chosen_assays</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dms_subset</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">dms_data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dms_data</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">chosen_assays</span><span class="op">]</span></span>
<span></span>
<span><span class="va">dms_subset</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">dms_subset</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metric_subset</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">metrics_scores</span><span class="op">[[</span><span class="st">"Spearman"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">DMS_ID</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">chosen_assays</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Number_of_Mutants</span>, <span class="va">Selection_Type</span>, <span class="va">UniProt_ID</span>, </span>
<span>        <span class="va">MSA_Neff_L_category</span>, <span class="va">Taxon</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">merge_am_dms</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="va">am_subset</span>, <span class="va">dms_subset</span>, </span>
<span>        by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DMS_id"</span> <span class="op">=</span> <span class="st">"DMS_id"</span>, <span class="st">"variant_id"</span> <span class="op">=</span> <span class="st">"mutant"</span><span class="op">)</span>,</span>
<span>        relationship <span class="op">=</span> <span class="st">"many-to-many"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>The Spearman scores are already provided for the 62 models in the
ProteinGym metrics table, but we will need to calculate this for
AlphaMissense.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spearman_res</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">merge_am_dms</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">DMS_id</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span></span>
<span>        AlphaMissense <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">AlphaMissense</span>, <span class="va">DMS_score</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        AlphaMissense <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">AlphaMissense</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">DMS_IDs</span> <span class="op">&lt;-</span> <span class="va">metric_subset</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">DMS_ID</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metric_subset</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">metric_subset</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">DMS_ID</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>DMS_id <span class="op">=</span> <span class="va">DMS_IDs</span><span class="op">)</span></span>
<span></span>
<span><span class="va">all_sp</span> <span class="op">&lt;-</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="va">metric_subset</span>, <span class="va">spearman_res</span>, </span>
<span>        by <span class="op">=</span> <span class="st">"DMS_id"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><code>all_sp</code> is a table of Spearman correlation values
(absolute values to handle difference in directionality of certain DMS
assays) across the ProteinGym and AlphaMissense models for the 10
assays.</p>
<p>Prepare the data for plotting.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res_long</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">all_sp</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">DMS_id</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>        cols <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html" class="external-link">everything</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        names_to <span class="op">=</span> <span class="st">"model"</span>, </span>
<span>        values_to <span class="op">=</span> <span class="st">"score"</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">model</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        model_mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        model <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">model</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">unique_ordered_models</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">res_long</span><span class="op">$</span><span class="va">model</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">res_long</span><span class="op">$</span><span class="va">model_mean</span>, </span>
<span>                                                        decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_long</span><span class="op">$</span><span class="va">model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">res_long</span><span class="op">$</span><span class="va">model</span>, </span>
<span>                   levels <span class="op">=</span> <span class="va">unique_ordered_models</span><span class="op">)</span></span>
<span></span>
<span><span class="va">topmodels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">res_long</span><span class="op">$</span><span class="va">model</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span></span>
<span><span class="va">top_spearmans</span> <span class="op">&lt;-</span> </span>
<span>    <span class="va">res_long</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">model</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">topmodels</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html" class="external-link">droplevels</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Visualize with a raincloud plot.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_spearmans</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">model</span>, y <span class="op">=</span> <span class="va">score</span>, fill <span class="op">=</span> <span class="va">model</span>, group <span class="op">=</span> <span class="va">model</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggdist</span><span class="fu">::</span><span class="fu"><a href="http://mjskay.github.io/ggdist/reference/stat_halfeye.html" class="external-link">stat_halfeye</a></span><span class="op">(</span></span>
<span>        adjust <span class="op">=</span> <span class="fl">.5</span>, </span>
<span>        width <span class="op">=</span> <span class="fl">.6</span>, </span>
<span>        .width <span class="op">=</span> <span class="fl">0</span>, </span>
<span>        justification <span class="op">=</span> <span class="op">-</span><span class="fl">.2</span>, </span>
<span>        point_colour <span class="op">=</span> <span class="cn">NA</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span></span>
<span>        width <span class="op">=</span> <span class="fl">.15</span>, </span>
<span>        outlier.shape <span class="op">=</span> <span class="cn">NA</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">gghalves</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gghalves/man/geom_half_point.html" class="external-link">geom_half_point</a></span><span class="op">(</span></span>
<span>        side <span class="op">=</span> <span class="st">"l"</span>, </span>
<span>        range_scale <span class="op">=</span> <span class="fl">.4</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fl">.2</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html" class="external-link">coord_cartesian</a></span><span class="op">(</span>clip <span class="op">=</span> <span class="st">"off"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_discrete.html" class="external-link">scale_fill_discrete</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Models"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Spearman"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span>, angle <span class="op">=</span> <span class="op">-</span><span class="fl">12</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p><img src="benchmarking_files/figure-html/raincloud-1.png" width="672"></p>
<p>Based on the Spearman correlation for the 10 assays we chose,
AlphaMissense performed the best. For a realistic and comprehensive
benchmark, one would want to apply this framework across all 217 DMS
assays available in ProteinGym.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2024-07-31 r86945)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin23.4.0</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.6.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Users/mtmorgan/bin/R-devel/lib/libRblas.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Users/mtmorgan/bin/R-devel/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] gghalves_0.1.4       ggplot2_3.5.1        ggdist_3.3.2        </span></span>
<span><span class="co">#&gt;  [4] tidyr_1.3.1          ExperimentHub_2.13.1 AnnotationHub_3.13.3</span></span>
<span><span class="co">#&gt;  [7] BiocFileCache_2.13.2 dbplyr_2.5.0         BiocGenerics_0.51.0 </span></span>
<span><span class="co">#&gt; [10] AlphaMissenseR_1.1.7 dplyr_1.1.4         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1        viridisLite_0.4.2       farver_2.1.2           </span></span>
<span><span class="co">#&gt;  [4] blob_1.2.4              filelock_1.0.3          Biostrings_2.73.1      </span></span>
<span><span class="co">#&gt;  [7] fastmap_1.2.0           duckdb_1.1.0            digest_0.6.37          </span></span>
<span><span class="co">#&gt; [10] mime_0.12               lifecycle_1.0.4         KEGGREST_1.45.1        </span></span>
<span><span class="co">#&gt; [13] RSQLite_2.3.7           magrittr_2.0.3          compiler_4.5.0         </span></span>
<span><span class="co">#&gt; [16] rlang_1.1.4             sass_0.4.9              tools_4.5.0            </span></span>
<span><span class="co">#&gt; [19] utf8_1.2.4              yaml_2.3.10             knitr_1.48             </span></span>
<span><span class="co">#&gt; [22] labeling_0.4.3          htmlwidgets_1.6.4       bit_4.5.0              </span></span>
<span><span class="co">#&gt; [25] spdl_0.0.5              curl_5.2.3              withr_3.0.1            </span></span>
<span><span class="co">#&gt; [28] purrr_1.0.2             desc_1.4.3              grid_4.5.0             </span></span>
<span><span class="co">#&gt; [31] stats4_4.5.0            fansi_1.0.6             colorspace_2.1-1       </span></span>
<span><span class="co">#&gt; [34] scales_1.3.0            cli_3.6.3               rmarkdown_2.28         </span></span>
<span><span class="co">#&gt; [37] crayon_1.5.3            ragg_1.3.2              generics_0.1.3         </span></span>
<span><span class="co">#&gt; [40] httr_1.4.7              BiocBaseUtils_1.7.3     DBI_1.2.3              </span></span>
<span><span class="co">#&gt; [43] cachem_1.1.0            zlibbioc_1.51.1         AnnotationDbi_1.67.0   </span></span>
<span><span class="co">#&gt; [46] BiocManager_1.30.23     XVector_0.45.0          vctrs_0.6.5            </span></span>
<span><span class="co">#&gt; [49] jsonlite_1.8.9          IRanges_2.39.2          S4Vectors_0.43.2       </span></span>
<span><span class="co">#&gt; [52] bit64_4.5.2             systemfonts_1.1.0       jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [55] glue_1.8.0              pkgdown_2.1.0           distributional_0.5.0   </span></span>
<span><span class="co">#&gt; [58] gtable_0.3.5            BiocVersion_3.20.0      GenomeInfoDb_1.41.1    </span></span>
<span><span class="co">#&gt; [61] UCSC.utils_1.1.0        munsell_0.5.1           tibble_3.2.1           </span></span>
<span><span class="co">#&gt; [64] pillar_1.9.0            rappdirs_0.3.3          htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [67] GenomeInfoDbData_1.2.12 R6_2.5.1                textshaping_0.4.0      </span></span>
<span><span class="co">#&gt; [70] evaluate_0.24.0         Biobase_2.65.1          highr_0.11             </span></span>
<span><span class="co">#&gt; [73] png_0.1-8               memoise_2.0.1           bslib_0.8.0            </span></span>
<span><span class="co">#&gt; [76] RcppSpdlog_0.0.18       rjsoncons_1.3.1         Rcpp_1.0.13            </span></span>
<span><span class="co">#&gt; [79] whisker_0.4.1           xfun_0.47               fs_1.6.4               </span></span>
<span><span class="co">#&gt; [82] pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martin Morgan, Tram Nguyen, Chan Zuckerberg Initiative DAF CZF2019-002443, NIH NCI ITCR U24CA180996, NIH NCI IOTN U24CA232979, NIH NCI ARTNet U24CA274159.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
